x-pack/test/security_solution_endpoint/apps/endpoint/endpoint_transform.ts

```diff
@@ -0,0 +1,147 @@
+
+    describe('united transform', () => {
+      const transformName = 'test-agg-united';
+      const indexA = 'test-agg-index-a';
+      const indexB = 'test-agg-index-b';
+      const transformDest = 'test-agg-destination';
+
+      before(async () => {
+        await esClient.indices.create({
+          index: indexA,
+          wait_for_active_shards: 1,
+          mappings: {
+            properties: {
+              '@timestamp': { type: 'date' },
+              uuid: { type: 'keyword' },
+              Endpoint: { type: 'keyword' },
+              a_value: { type: 'keyword' },
+            },
+          },
+        });
+        await esClient.indices.create({
+          index: indexB,
+          wait_for_active_shards: 1,
+          mappings: {
+            properties: {
+              '@timestamp': { type: 'date' },
+              uuid: { type: 'keyword' },
+              agent: { type: 'keyword' },
+              b_value: { type: 'keyword' },
+            },
+          },
+        });
+        await esClient.transform.putTransform({
+          transform_id: transformName,
+          dest: {
+            index: transformDest,
+          },
+          source: {
+            index: [indexA, indexB],
+          },
+          frequency: '1s',
+          sync: {
+            time: {
+              delay: '3s',
+              field: '@timestamp',
+            },
+          },
+          pivot: {
+            aggs: {
+              join: {
+                scripted_metric: transformAggregation,
+              },
+            },
+            group_by: {
+              uuid: {
+                terms: {
+                  field: 'uuid',
+                },
+              },
+            },
+          },
+        });
+      });
+      after(async () => {
+        await endpointTestResources.stopTransform(transformName);
+        await esClient.transform.deleteTransform({
+          transform_id: transformName,
+          force: true,
+          delete_dest_index: true,
+        });
+        await esClient.indices.delete({ index: indexA });
+        await esClient.indices.delete({ index: indexB });
+      });
+
+      it('runs scripted metric agg to join data', async () => {
+        await esClient.index({
+          index: indexA,
+          refresh: 'wait_for',
+          document: {
+            '@timestamp': new Date().getTime(),
+            uuid: '8000',
+            Endpoint: 'e', // Endpoint key must be present. value does not matter
+            a_value: 'A', // key to test for
+          },
+        });
+        await esClient.index({
+          index: indexB,
+          refresh: 'wait_for',
+          document: {
+            '@timestamp': new Date().getTime(),
+            uuid: '8000',
+            agent: 'a', // agent key must be present
+            b_value: 'B', // key to test for
+          },
+        });
+        await endpointTestResources.startTransform(transformName);
+
+        await sleep(1000);
```

Comment:
Is this `sleep()` necessary given we are doing a retry in the next line?

Reply:
not strictly necessary, no. Could move this second to add one more to the retry timeout. But this certainly won't work for the first second or two

Reply:
I would add it to the retry timeout, we don't like these explicit waits in our tests :D

Reply:
I second @szwarckonrad 's suggestion. We should try not to use "sleep"s in our tests if at all possible.